### PR TITLE
Feature: dynamic required rule

### PR DESCRIPTION
**Pull request by indrimuska**
It allows to set `required` rule as function (not only a bool or an object) like this:
```javascript
rules: {
   myInput: {
      required: function (myInputValue) {
         // .. perform your checks
         return true;
      }
   }
}
```

This is an example: http://plnkr.co/edit/b45TxES4hnYAlewxK4AX?p=preview

### CODE REVIEW

File: source/services/model-validator.ts [hunk 1]
```diff
@@ -474,9 +474,11 @@ module formFor {
 
     private validateFieldRequired_(value:any, validationRules:ValidationRules):any {
       if (validationRules.required) {
-        var required:boolean = angular.isObject(validationRules.required) ?
-          (<ValidationRuleBoolean> validationRules.required).rule :
-          <boolean> validationRules.required;
+        var required:boolean = angular.isObject(validationRules.required)
+          ? (<ValidationRuleBoolean> validationRules.required).rule
+          : angular.isFunction(validationRules.required)
+            ? validationRules.required.call(this, value)
```
Comment (bvaughn) [inline source/services/model-validator.ts]: This requires an update to `ValidationRules` to change...
```required?:boolean|ValidationRuleBoolean;```
...to...
```required?:boolean|ValidationRuleBoolean|Function;```

Comment (bvaughn) [inline source/services/model-validator.ts]: Nevermind I'll just add that.

**Comment by bvaughn** [comment]
This change-set LGTM to me as well. Thanks for making these changes man. Can you update the `gh-page` docs?

**Comment by indrimuska** [comment]
Ok, no problem. I will do this today.

**Comment by indrimuska** [comment]
I'm not sure to know where is the doc page for the `required` rule, maybe [this](http://bvaughn.github.io/angular-form-for/#/demo/validation-types)?

Anyway, I think minimum/maximum/required rules needs to be updated - when their type is `Function` - in order to give to the users not only the field value as parameter, but also the name and the entire model data, as in the `custom` rule (lines: [147](https://github.com/bvaughn/angular-form-for/blob/master/source/services/model-validator.ts#L147), [303](https://github.com/bvaughn/angular-form-for/blob/master/source/services/model-validator.ts#L303)).

What are your thought about that?

**Comment by bvaughn** [comment]
Oh sorry, I missed this comment.

I'd say... in order to avoid breaking backwards compatibility (and having to do a major version bump) we might want to pass it the params in the order of `(value, validationRules, formData, fieldName)`

**Comment by bvaughn** [comment]
And yeah, you're right about that being the docs page. It's pretty sparse, I guess. :/

**Comment by indrimuska** [comment]
Ok, don't worry, I'll make those changes tomorrow! :wink:

**Pull request merged**